examples/toolchains/cc/README.md

```diff
@@ -0,0 +1,17 @@
+C++ Toolchain Example
+=====================
+
+This is an example C++ project that uses `rules_cc`. It should build and run both with or without Nix installed.
+
+# Usage
+
+To run the example with Nix, issue the following command:
+```
+nix-shell --command 'bazel run :hello'
+```
+
+To run the example without Nix, make sure you have Bazel installed, and issue the following command:
+```
+bazel run :hello
+```
+Note that if this command is run on NixOS, it will still use Nix. :-)
```

Comment:
Assuming the `build:nix` syntax in `.bazelrc`, i.e. such that the `--host_platform` is *not* set to the Nix one, this should attempt to use the `local_config_cc` toolchain instead of the Nix provided one. I don't remember if that would work or break on NixOS. That toolchain doesn't download any binaries, it just tries to find tools in `PATH`. If those happen to exist in the given NixOS environment, then this may well still work. But, it's not going to be as hermetic as the Nix provided toolchain.

The same kind of issue holds in the Go case. Though the default Go toolchain will download binaries from the web. If those are statically linked then they may well work on NixOS. But, they would not be Nix provided tools.